### PR TITLE
legacy storeapi: use craft-store

### PR DESCRIPTION
**Pull request by sergiusens**
- Replace the http_clients implementation in storeapi with craft-store.
- Remove no longer necessary tests.
- Adapt the code base to the slight API changes (with consideration of
- being entirely replaced as commands migrate out of legacy).

Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: snapcraft_legacy/_store.py [hunk 1]
```diff
@@ -807,8 +825,9 @@ def _upload_delta(
             raise storeapi.errors.StoreDeltaApplicationError(str(e))
         else:
             raise
-    except storeapi.http_clients.errors.StoreServerError as e:
-        raise storeapi.errors.StoreUploadError(snap_name, e.response)
+    except craft_store.errors.StoreServerError as store_error:
+        if store_error.response.status_code == 401:
```
Comment (sergiusens) [inline snapcraft_legacy/_store.py:829]: this is wrong

Comment (cmatsuoka) [reply snapcraft_legacy/_store.py:829]: Store returning wrong status or wrong logic testing status?

**Review by cmatsuoka — COMMENTED** [review]
Didn't find obvious coding errors. I'm assuming the protocol implementation is correct, and pbr-related errors should be fixed once #3688 is merged in, but we still have many keyring-related failures in the environment tests.

File: snapcraft_legacy/_store.py (hunk 1 above)
Comment (sergiusens) [reply snapcraft_legacy/_store.py:829]: it is the wrong logic :-)

**Comment by sergiusens** [comment]
Favoring #3692

**Pull request closed without merging**